### PR TITLE
[Feature]Implement dump results

### PR DESCRIPTION
**Pull request by wanghonglie**
Implement dump results

### CODE REVIEW

**Review by chhluo — DISMISSED** [review]

File: tools/test.py [hunk 1]
```diff
@@ -97,6 +101,16 @@ def main():
...
         cfg = trigger_visualization_hook(cfg, args)
 
+    # Dump predictions
+    if args.dump is not None:
+        assert args.dump.endswith(('.pkl', '.pickle')), \
+            'The dump file must be a pkl file.'
+        dump_metric = dict(type='DumpResults', out_file_path=args.dump)
+        if isinstance(cfg.test_evaluator, (list, tuple)):
+            cfg.test_evaluator = list(cfg.test_evaluator).append(dump_metric)
+        else:
```
Comment (ZwwWayne) [inline tools/test.py:111]: What if the evaulator has a type and the metric is wrapped inside like
```python
test_evaluator = dict(
    type='xxxevaluator',
    metric=[dict(type='CocoMetric'), ])
```

**Review by RangiLyu — DISMISSED** [review]

File: tools/test.py [hunk 2]
```diff
@@ -19,6 +19,10 @@ def parse_args():
     parser.add_argument(
         '--work-dir',
         help='the directory to save the file containing evaluation metrics')
+    parser.add_argument(
+        '--dump',
```
Comment (ZwwWayne) [inline tools/test.py:23]: Consider to use `--format-only` or some keys that exists in MMDet 2.x for compatibility.

File: tools/test.py [hunk 3]
```diff
@@ -97,6 +101,25 @@ def main():
     if args.show or args.show_dir:
         cfg = trigger_visualization_hook(cfg, args)
 
+    # Dump predictions
+    if args.dump is not None:
+        assert args.dump.endswith(('.pkl', '.pickle')), \
```
Comment (ZwwWayne) [inline tools/test.py:106]: Try to implement a function to encapsulate lines 106-121

File: tools/test.py (hunk 2 above)
Comment (RangiLyu) [reply tools/test.py:23]: We follow the MMCls's args. Do we have any conventions?

**Review by ZwwWayne — APPROVED** [review]

**Review by ZwwWayne — APPROVED** [review]

**Pull request merged**